src/DebugChannel.js

```diff
@@ -42,14 +44,36 @@ export class DebugChannel {
   /* Request object based on the protocol
   */
   readIn(): string {
-    let contents = "";
-    while (contents.length === 0) {
-      contents = this.fs.readFileSync(this.inFilePath, "utf8");
+    let message = "";
+    while (true) {
```

Comment:
Most of the high level logic are dealing with partial reading. Please add a big comment explaining this high level goal here.